### PR TITLE
provide the guid when creating content

### LINKED ISSUE #859: Failure to add environment variables to content
When publishing a new content item with environment variables, the initial API call to deploy the content succeeds, but the secondary call to add environment variables fails with a 404 error.

I saw this both with a local dev Connect environment and with a proper instance. Logs from the local dev example attached.

```
Completed deployed execution
── Deployment complete ─────────────────────────────────────────────────────────────────────────────────────────────
✔ Successfully deployed to <http://localhost:3939/content/6441c48e-771b-4db6-ab58-fda27a036386/>
> rsconnect::deployDoc("index.Rmd", appName = "rmd-with-secret", envVars = c(SECRET = "forty-two"))
ℹ Discovering document dependencies...
✔ Document dependencies discovered
── Preparing for deployment ────────────────────────────────────────────────────────────────────────────────────────
Found multiple accounts.
Which one do you want to use?
1: server: localhost / username: toph
2: server: rsc.radixu.com / username: toph
Selection: 1
✔ Deploying "rmd-with-secret" to "server: localhost / username: toph"
ℹ Creating application on server...
✔ Created application with id 525
ℹ Updating environment variables forty-two...
Error in `PATCH()`:
! <http://localhost:3939/__api__> failed with HTTP status 404
404 page not found
Run `rlang::last_trace()` to see where the error occurred.
```

### PR DESCRIPTION
**Pull request by aronatkins**
fixes #859

### CODE REVIEW

File: R/deployApp.R [hunk 1]
```diff
@@ -361,7 +361,7 @@ deployApp <- function(appDir = getwd(),
     )
     taskComplete(quiet, "Created application with id {.val {application$id}}")
   } else {
-    application <- taskStart(quiet, "Looking up application with id {.val {target$appId}}...")
+    taskStart(quiet, "Looking up application with id {.val {target$appId}}...")
```
Comment (hadley) [inline R/deployApp.R:364]: I don't see `guid` used here, but I assume it is further down.

File: R/client-connect.R [hunk 2]
```diff
@@ -59,6 +59,7 @@ connectClient <- function(service, authInfo) {
       application <- POST_JSON(service, authInfo, "/applications", details)
       list(
         id = application$id,
+        guid = application$guid,
```
Comment (hadley) [inline R/client-connect.R:62]: Need same change for other clients?

**Review by hadley — APPROVED** [review]

File: R/deployApp.R (hunk 1 above)
Comment (aronatkins) [reply R/deployApp.R:364]: This was an unrelated mistake; we double-assigned `application` but the first assignment (from `taskStart`) was incorrect and immediately overwritten by the next statement. Found this while trying to trace the definition of `application`.

File: R/client-connect.R (hunk 2 above)
Comment (aronatkins) [reply R/client-connect.R:62]: I don't think so. Only connect has `guid`. See the set of fields that are "kept" from API list responses in `applications()`.

**Pull request merged**